doc/source/whatsnew/v1.4.0.rst

```diff
@@ -509,7 +509,7 @@ Conversion
 
 Strings
 ^^^^^^^
--
+- Fixed bug in :bug:`is_dtype_equal` raising if pyarrow not installed (:issue:`44327`)
```

Comment:
this won't render & is not public facing anyhow. remove the note

Reply:
I did not understand the part where "is not public facing anyhow.". I will anyhow remove it.

Reply:
Do I remove this?

Reply:
I think you can keep it, but reword it a bit to not mention `is_dtype_equal` explicitly. Something like "Fixed a bug in checking if a dtype is the `string[pyarrow]` dtype incorrectly raising an error when pyarrow is not installed (:issue:...)"

Reply:
Done